pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -105,6 +111,96 @@ const (
 	IngressReady apis.ConditionType = "IngressReady"
 )
 
+type ModelStatus struct {
+	// Whether the available predictor endpoint reflects the current Spec or is in transition
+	// +kubebuilder:default=UpToDate
+	TransitionStatus TransitionStatus `json:"transitionStatus"`
+
+	// High level state string: Pending, Standby, Loading, Loaded, FailedToLoad
+	// +kubebuilder:default=Pending
+	ActiveModelState ModelState `json:"activeModelState"`
+	// +kubebuilder:default=""
+	TargetModelState ModelState `json:"targetModelState"`
+
+	// Details of last failure, when load of target model is failed or blocked
+	// +optional
+	LastFailureInfo *FailureInfo `json:"lastFailureInfo,omitempty"`
+
+	// How many copies of this predictor's models failed to load recently
+	// +kubebuilder:default=0
+	FailedCopies int `json:"failedCopies"`
+}
```

Comment:
I found that the modelmesh predictors' status output from `kubectl describe` is kind of jumbled because there's no way to control the order of printed fields AFAIK.

So I was thinking that it would be good to group/nest some of the fields within other structs, specifically:
- Active and target model states
- Model copy counts assuming that we add another per the comment above

Reply:
Any recommendation for the json tag naming for these structs?